ql/src/java/org/apache/hadoop/hive/ql/ddl/table/drop/DropTableDesc.java

```diff
@@ -29,31 +32,40 @@
  * DDL task description for DROP TABLE commands.
  */
 @Explain(displayName = "Drop Table", explainLevels = { Level.USER, Level.DEFAULT, Level.EXTENDED })
-public class DropTableDesc implements DDLDesc, Serializable {
+public class DropTableDesc implements DDLDescWithWriteId, Serializable {
   private static final long serialVersionUID = 1L;
 
-  private final String tableName;
+  private final TableName tableName;
   private final boolean ifExists;
   private final boolean purge;
   private final ReplicationSpec replicationSpec;
   private final boolean validationRequired;
+  private final boolean isTransactional;
 
-  public DropTableDesc(String tableName, boolean ifExists, boolean ifPurge, ReplicationSpec replicationSpec) {
-    this(tableName, ifExists, ifPurge, replicationSpec, true);
+  private long writeId = 0;
+
+  public DropTableDesc(TableName tableName, boolean ifExists, boolean ifPurge, ReplicationSpec replicationSpec) {
+    this(tableName, ifExists, ifPurge, replicationSpec, true, null);
   }
 
-  public DropTableDesc(String tableName, boolean ifExists, boolean purge, ReplicationSpec replicationSpec,
+  public DropTableDesc(TableName tableName, boolean ifExists, boolean ifPurge, ReplicationSpec replicationSpec,
```

Comment:
Nit: the variable name should be purge - I know, it was ifPurge before, but it's a great opportunity to fix it :)